src/quadratureMethods/momentInversion/univariate/basic/hyperbolicMomentInversion/hyperbolicMomentInversion.C

```diff
@@ -0,0 +1,272 @@
+              - 3.0*moments[0]*moments[1]*moments[2]
+              + 2.0*Foam::pow3(moments[1])
+            )/Foam::pow3(moments[0]);
+
+        centralMoments_[4] =
+            (
+                Foam::pow3(moments[0])*moments[4]
+              - 4.0*Foam::sqr(moments[0])*moments[1]*moments[3]
+              + 6.0*moments[0]*Foam::sqr(moments[1])*moments[2]
+              - 3.0*Foam::pow4(moments[1])
+            )/Foam::pow4(moments[0]);
+
+        centralMoments_[5] =
+            centralMoments_[3]/Foam::sqr(centralMoments_[2])
+           /(
+                2.0*centralMoments_[2]*centralMoments_[4]
+              - Foam::sqr(centralMoments_[3])
+            );
+    }
+}
+
+void Foam::hyperbolicMomentInversion::unscaleAbscissae
+(
+    const scalarList& moments
+)
+{
+    scalar m0 = moments[0];
+    scalar uBar = moments[1]/moments[0];
+
+    forAll(abscissae_, nodei)
+    {
+        abscissae_[nodei] += uBar;
+        weights_[nodei] *= m0;
+    }
+}
+// * * * * * * * * * * * * * * * * Constructors  * * * * * * * * * * * * * * //
+
+Foam::hyperbolicMomentInversion::hyperbolicMomentInversion
+(
+    const dictionary& dict
+)
+:
+    univariateMomentInversion(dict),
+    centralMoments_(0)
+{}
+
+
+// * * * * * * * * * * * * * * * * Destructor  * * * * * * * * * * * * * * * //
+
+Foam::hyperbolicMomentInversion::~hyperbolicMomentInversion()
+{}
+
+
+// * * * * * * * * * * * * * * * Member Functions  * * * * * * * * * * * * * //
+
+void Foam::hyperbolicMomentInversion::correctRecurrence
+(
+    univariateMomentSet& moments,
+    scalarList& alpha,
+    scalarList& beta,
+    const scalar minKnownAbscissa,
+    const scalar maxKnownAbscissa
+)
+{
+    return; // No correction needed for Gauss quadrature
+}
+
+
+void Foam::hyperbolicMomentInversion::calcNQuadratureNodes
+(
+    univariateMomentSet& moments
+)
+{
+    label nRealizableMoments = moments.nRealizableMoments();
+
+    if (nRealizableMoments >= 2)
+    {
+        if (nRealizableMoments % 2 != 0)
+        {
+            nInvertibleMoments_ = nRealizableMoments - 1;
+        }
+        else
+        {
+            nInvertibleMoments_ = nRealizableMoments;
+        }
+    }
+    else
+    {
+        FatalErrorInFunction
+            << "The moment set has size less or equal to 1." << nl
+            << "    Moment set: " << moments
+            << abort(FatalError);
+    }
+
+    nNodes_ = nInvertibleMoments_/2;
+
+    if (nNodes_ > 3)
+    {
+        FatalErrorInFunction
+            << "Hyperbolic moment inversion can handle a maximum of " << nl
+            << "3 nodes. " << nNodes_ << " were selected."
+            << abort(FatalError);
+    }
+
+    abscissae_.setSize(nNodes_);
+    weights_.setSize(nNodes_);
+}
+
+
+void Foam::hyperbolicMomentInversion::invert
+(
+    univariateMomentSet& moments,
+    const scalar minKnownAbscissa,
+    const scalar maxKnownAbscissa
+)
+{
+    calcNQuadratureNodes(moments);
+
+    centralMoments_ = scalarList(moments.size(), scalar(0));
```

Comment:
Replace with centralMoments_.setSize(moments.size(), 0);

Also, if possible, only allocate the actual number of central moments needed for the inversion due to realizability.